src/tools/gophish_export.py

```diff
@@ -253,6 +253,44 @@ def get_application(rawEvent):
     return application
 
 
+def export_user_reports(api, assessment_id):
+    """Build and export a user_report JSON object for each campaign in an assessment."""
+    campaign_ids = get_campaign_ids(api, assessment_id)
+
+    for campaign_id in campaign_ids:
+        first_report = None
+        user_report_doc = dict()
+        campaign = get_campaign_data(api, campaign_id)
+
+        # iterate over clicks and find the earliest click
+        for click in campaign["clicks"]:
+            if (
+                first_report is None
+                or datetime.strptime(click["time"].split(".")[0], "%Y-%m-%dT%H:%M:%S")
+                < first_report
+            ):
+                first_report = datetime.strptime(
+                    click["time"].split(".")[0], "%Y-%m-%dT%H:%M:%S"
+                )
+
+        user_report_doc["customer"] = None
+        user_report_doc["assessment"] = assessment_id
+        user_report_doc["campaign"] = campaign["id"]
```

Comment:
Is there any difference between `campaign_id` and `campaign["id"]` here?  If not, I'd simply go with:
```suggestion
        user_report_doc["campaign"] = campaign_id
```

Reply:
I tried that before during my testing. The campaign_id that iterates over campaign_ids, which is in turn returned from get_campaign_ids() is an integer value. campaign["id"] is a string value. The schema for the campaign field in the user_report is defined as a string as well, as thats what campaign _ids were defined as throughout the PCA database and our legacy data. Im using the campaign["id"] so I can load string data into the current schema.

Reply:
Thanks for the explanation.  In the interest of more readable code, how about something like this?
```
# get_campaign_ids() returns integers, but user_report_doc["campaign"]
# expects a string
user_report_doc["campaign"] = str(campaign_id)
```

Reply:
See 724b35ac39da24c61713cc0dd0addb7a5f3befe5